analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotTestThisWithIsOperator.cs

```diff
@@ -19,38 +19,111 @@
         private const string MessageFormat = "Offload the code that's conditional on this 'is' test to the appropriate subclass and remove the test.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
         protected override void Initialize(SonarAnalysisContext context)
         {
-            context.RegisterSyntaxNodeActionInNonGenerated(c =>
-                {
-                    var expression = (BinaryExpressionSyntax)c.Node;
-                    if (expression.Left is ThisExpressionSyntax)
-                    {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, expression.GetLocation()));
-                    }
-                },
-                SyntaxKind.IsExpression);
+            context.RegisterSyntaxNodeActionInNonGenerated(AnalyzeIsExpression, SyntaxKind.IsExpression);
+            context.RegisterSyntaxNodeActionInNonGenerated(AnalyzeIsPatternExpression, SyntaxKindEx.IsPatternExpression);
+            context.RegisterSyntaxNodeActionInNonGenerated(AnalyzeSwitchExpression, SyntaxKindEx.SwitchExpression);
+            context.RegisterSyntaxNodeActionInNonGenerated(AnalyzeSwitchStatement, SyntaxKind.SwitchStatement);
         }
+
+        private static void AnalyzeIsExpression(SyntaxNodeAnalysisContext context)
+        {
+            if (((BinaryExpressionSyntax)context.Node).Left.RemoveParentheses() is ThisExpressionSyntax)
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, context.Node.GetLocation()));
+            }
+        }
+
+        private static void AnalyzeIsPatternExpression(SyntaxNodeAnalysisContext context)
+        {
+            var isPatternExpression = (IsPatternExpressionSyntaxWrapper)context.Node;
+            if (isPatternExpression.Expression.RemoveParentheses() is ThisExpressionSyntax
+                && ContainsTypeCheckInPattern(isPatternExpression))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, context.Node.GetLocation()));
+            }
+        }
+
+        private static void AnalyzeSwitchStatement(SyntaxNodeAnalysisContext context)
+        {
+            var switchStatement = (SwitchStatementSyntax)context.Node;
+            if (switchStatement.Expression.RemoveParentheses() is ThisExpressionSyntax
+                && (ContainsTypeCheckInPattern(switchStatement)
+                    || ContainsTypeCheckInCaswSwitchLabel(switchStatement)))
```

Comment:
Let's make this single line, the EOL is hard to read there
```suggestion
                && (ContainsTypeCheckInPattern(switchStatement) || ContainsTypeCheckInCaswSwitchLabel(switchStatement)))
```

Reply:
After the last changes I think it is not applicable.